synapse/storage/util/id_generators.py

```diff
@@ -232,7 +232,7 @@ def __init__(
         # that allows us to skip forwards when there are gapless runs of
         # positions.
         self._persisted_upto_position = (
-            min(self._current_positions.values()) if self._current_positions else 0
+            min(self._current_positions.values()) if self._current_positions else 1
```

Comment:
I do not understand this change.

Reply:
Oh woops. I forgot about this one too. Basically 1) if you try and use a stream ordering of 0 then things go bang and 2) the stream orderings turn out to start at 2, so it makes sense to use 1 as the initial value.

Will add a comment about choice of 1

Reply:
Looks better. Thanks!